extensions/panache/panache-mock/runtime/src/main/java/io/quarkus/panache/mock/PanacheMock.java

```diff
@@ -0,0 +1,81 @@
+package io.quarkus.panache.mock;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.mockito.Mockito;
+import org.mockito.internal.invocation.DefaultInvocationFactory;
+import org.mockito.internal.invocation.InterceptedInvocation;
+import org.mockito.internal.invocation.RealMethod;
+import org.mockito.internal.util.MockUtil;
+import org.mockito.invocation.MockHandler;
+import org.mockito.mock.MockCreationSettings;
+
+public class PanacheMock {
+
+    public static boolean IsMockEnabled = false;
+
+    private final static Map<Class<?>, Object> mocks = new HashMap<>();
```

Comment:
Maybe use a ConcurentHashMap in case tests are run in parallel ?

Reply:
That would not help: tests can't run in parallel with that setup, because the set of classes to mock and the methods mocked is test-specific. So we'd have to make this a thread-local, which would mean that reactive code would likely fail or require context propagation.

Reply:
I think it's OK for a first implementation as soon as it is clearly documented that test with panache mock should not be run in parallel.

Reply:
Is it possible that running user code will access this? e.g. If I mock a request then send a HTTP request will the HTTP request handler code access this map? If so it should be a CHM anyway to make sure that any modifications are visible to the other thread.

Reply:
Fixed